### PR TITLE
Expand the valid CSS Identifier method to allow more valid characters in class names

### PR DESCRIPTION
**Pull request by soluml**
**Summary**

The hyphen (or dash) character is a valid character for CSS class names (under most circumstances) and does not need to be escaped. Characters from the Latin-1 Supplement unicode character range through the Specials unicode range are also valid and do not need to be escaped.

This PR expands the regex used to filter out invalid characters to allow these.

**Test plan**

There are no tests for this method currently. However, we can observe that these characters are valid for CSS class names by [looking at the spec](https://www.w3.org/TR/CSS2/syndata.html#characters). You can [read more information about the additional unicode ranges added](https://jrgraphix.net/research/unicode_blocks.php), and see a read a fantastic write up about [constructing class names here](https://mathiasbynens.be/notes/css-escapes).

This PR will address issue #554.

### CODE REVIEW

File: src/__tests__/__snapshots__/babel.test.ts.snap [hunk 1]
```diff
@@ -1,559 +1,559 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Anber) [inline src/__tests__/__snapshots__/babel.test.ts.snap:1]: It looks like your snapshot has a different line ending symbol. Could you please check it?

**Comment by jayu** [comment]
Let me check if this does not break my example project builds ^^ I think we should add example project building to our pipelines

**Comment by Anber** [comment]
@jayu we can use http://linaria.now.sh/ as an example project :)

**Comment by jayu** [comment]
@Anber It's a great idea but we might want to have better separation of concerns. I opened an issue for that #565

**Comment by jayu** [comment]
Just checked and it works for me. @soluml  please address @Anber comment and we are ready to go.

File: src/__tests__/__snapshots__/babel.test.ts.snap (hunk 1 above)
Comment (soluml) [reply src/__tests__/__snapshots__/babel.test.ts.snap]: Yep, my bad! Editor didn't recognize this as a LF project. Updated!

**Comment by soluml** [comment]
@jayu Line Endings for the .snap file have been resolved.

**Pull request merged**